### PR TITLE
[Asset snapshotting] Correctly handle all cases of serving HTML files

### PR DESCRIPTION
**Pull request by Que3216**

### CODE REVIEW

File: packages/replayer/src/assets/snapshot-assets.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { mkdir, writeFile } from "fs/promises";
-import { dirname, join } from "path";
+import path, { dirname, extname, join } from "path";
```
Comment (leafty) [inline packages/replayer/src/assets/snapshot-assets.ts:2]: Apparently imports should be optimized here.

**Review by leafty — APPROVED** [review]
LGTM

Note: it feels that this code is doing exactly what a server dedicated to serving single-page apps would do.

File: packages/replayer/src/assets/snapshot-assets.ts (hunk 1 above)
Comment (Que3216) [reply packages/replayer/src/assets/snapshot-assets.ts:2]: Ah yes! Not sure how that slipped in. I'll fix.

Comment (Que3216) [reply packages/replayer/src/assets/snapshot-assets.ts:2]:
```suggestion
import { dirname, extname, join } from "path";
```

**Pull request merged**